autoload/ale/util.vim

```diff
@@ -90,6 +90,17 @@ function! ale#util#LocItemCompare(left, right) abort
         return 1
     endif
 
+    " put warnings after errors (for the same line) since the text that shows
+    " when the cursor is moved will show only the first entry
+
+    if a:left['type'] < a:right['type']
+      return -1
+    endif
+
+    if a:left['type'] > a:right['type']
+      return 1
+    endif
```

Comment:
If you're going to sort on the error type, you should take issue #149 into account. Eventually, I would like to support error types in some way beyond "warning" and "error". 'E' seems to be the first single character which could appear in lexicographic order, but 'I' comes before 'W' in lexicographic order, and 'W' is more severe than 'I', so that wouldn't be right.